packages/bundler/src/BundlerCollectorTracer.ts

```diff
@@ -159,28 +175,47 @@ export function bundlerCollectorTracer (): BundlerCollectorTracer {
         }
       }
 
+      if (log.getDepth() === 1) {
+        if (opcode === 'CALL' || opcode === 'STATICCALL') {
+          // stack.peek(0) - gas
+          const addr = toAddress(log.stack.peek(1).toString(16))
+          const topLevelTargetAddress = toHex(addr)
+          // stack.peek(2) - value
+          const ofs = parseInt(log.stack.peek(3).toString())
+          let len = parseInt(log.stack.peek(4).toString())
+          if (len > 4) len = 4
+          const topLevelMethodSig = toHex(log.memory.slice(ofs, ofs + len))
+
+          this.currentLevel = this.topLevelCalls[this.topLevelCallCounter] = {
+            topLevelMethodSig,
+            topLevelTargetAddress,
+            access: {},
+            opcodes: {},
+            contractSize: {}
+          }
+          this.topLevelCallCounter++
+        } else if (opcode === 'LOG1') {
+          // ignore log data ofs, len
+          const topic = log.stack.peek(2).toString(16)
+          if (topic === this.stopCollectingTopic) {
+            this.stopCollecting = true
+          }
+        }
+        this.lastOp = ''
+        return
+      }
+
       if (opcode.match(/^(EXT.*|CALL|CALLCODE|DELEGATECALL|STATICCALL)$/) != null) {
         // this.debug.push('op=' + opcode + ' last=' + this.lastOp + ' stacksize=' + log.stack.length())
         const idx = opcode.startsWith('EXT') ? 0 : 1
         const addr = toAddress(log.stack.peek(idx).toString(16))
         const addrHex = toHex(addr)
         if ((this.currentLevel.contractSize[addrHex] ?? 0) === 0 && !isPrecompiled(addr)) {
           this.currentLevel.contractSize[addrHex] = db.getCode(addr).length
-        }
-      }
-
-      if (log.getDepth() === 1) {
-        // NUMBER opcode at top level split levels
-        if (opcode === 'NUMBER') this.numberCounter++
-        if (this.numberLevels[this.numberCounter] == null) {
-          this.currentLevel = this.numberLevels[this.numberCounter] = {
-            access: {},
-            opcodes: {},
-            contractSize: {}
+          if (this.currentLevel.contractSize[addrHex] === 0) {
+            this.debug.push(`empty addr: ${opcode} ${addrHex}`)
```

Comment:
I don't understand this change. Isn't the empty address call forbidden?

Reply:
sorry. its debug leftover, which was fixed.